core/db-install/src/main/java/org/opennms/core/db/install/InstallerDb.java

```diff
@@ -2322,4 +2323,25 @@ public void vacuumDatabase(final boolean full) throws SQLException {
             m_out.println("OK");
         }
     }
+
+    // Ensures that the database time and the system time running the installer match
+    // If the difference is greater than 1s, it fails
+    public void checkTime() throws Exception {
+        m_out.print("- checking if time of database \"" + getDatabaseName() + "\" is matching system time... ");
+
+        try (Statement st = getConnection().createStatement()) {
+            try (ResultSet rs = st.executeQuery("SELECT NOW()")) {
+                if (rs.next()) {
+                    final Timestamp currentDatabaseTime = rs.getTimestamp(1);
+                    final long currentSystemTime = System.currentTimeMillis();
+                    final long diff = currentDatabaseTime.getTime() - currentSystemTime;
+                    if (Math.abs(diff) > 1000) {
+                        m_out.println("NOT OK");
+                        throw new Exception("Database time and system time differ. Please update either the database time or the system time");
```

Comment:
It might be a good idea to include the (formatted) timestamps and the delta in the exception message.